JavaScript/JavaScriptSDK.Tests/CheckinTests/PageVisitTimeManager.tests.ts

```diff
@@ -0,0 +1,185 @@
+﻿/// <reference path="..\TestFramework\Common.ts" />
+/// <reference path="../../JavaScriptSDK/Telemetry/PageVisitTimeManager.ts" />
+class PageVisitTimeManagerTests extends TestClass {
+
+    private throwInternalNonUserActionableSpy;
+    private throwInternalUserActionableSpy;
+    private getStorageObjectStub;
+
+    /** Method called before the start of each test method */
+    public testInitialize() {
+        var storage = this.getMockStorage();
+        this.getStorageObjectStub = sinon.stub(Microsoft.ApplicationInsights.Util, "_getSessionStorageObject",() => storage);
+
+        this.throwInternalNonUserActionableSpy = sinon.spy(Microsoft.ApplicationInsights._InternalLogging, "throwInternalNonUserActionable");
+        this.throwInternalUserActionableSpy = sinon.spy(Microsoft.ApplicationInsights._InternalLogging, "throwInternalUserActionable");
+    }
+
+    /** Method called after each test method has completed */
+    public testCleanup() {
+        this.getStorageObjectStub.restore();
+        this.throwInternalNonUserActionableSpy.restore();
+        this.throwInternalUserActionableSpy.restore();
+    }
+
+    public registerTests() {
+
+        var testValues = {
+            page1Name: "page1",
+            page1Url: "page1Url",
+            page1ViewTime: 10,
+
+            page2Name: "page2",
+            page2Url: "page2Url",
+            page2ViewTime: 20,
+
+            page3Name: "page3",
+            page3Url: "page3Url",
+            page3ViewTime: 30
+        };
+
+        this.testCase({
+            name: "PageVisitTimeManager: When trackPreviousPageVisit is called once, the tracking delegate is not called since there are no previous pages",
+            test: () => {
+
+                //setup
+                var object = { method: function () { } };
+                var spy = sinon.spy(object, "method");
```

Comment:
a better naming for "method" might help to understand its purpose in the test (like "trackPageVisitCallbackStub")